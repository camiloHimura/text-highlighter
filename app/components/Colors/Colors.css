.colorsContainer{
    display: flex;
}

.colorsContainer > div{
    width: 60px;
    height: 40px;
    cursor: pointer;
    margin-left: 10px;
    border-radius: 2px;
    transition: all .2s;
    border: 1px solid black;
}

.colorsContainer > div.selected{
    width: 40px;
    height: 40px;
    border-radius: 50%;
}

.colorsContainer h2{
    margin-left: 20px;
}
