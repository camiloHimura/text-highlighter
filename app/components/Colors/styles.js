import styled from 'styled-components'

export const Container = styled.div`
  width: 60px;
  height: 40px;
  cursor: pointer;
  margin-left: 10px;
  border-radius: 2px;
  transition: all .2s;
  border: 1px solid black;
  
  &.selected{
    width: 40px;
    height: 40px;
    border-radius: 50%;
  }
`


export const Title = styled.h2`  
  margin-left: 20px;
`

