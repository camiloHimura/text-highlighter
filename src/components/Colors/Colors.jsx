import React from 'react';
import "./Colors.css"

function Colors(props){
    const {options, optionSelected, onClick, title = ''} = props;

    return  <div className="colorsContainer --flex --alignCenter">
                {options.map((color, key) => {
                    const selected = optionSelected === color ? 'selected': '';

                    return <div key={`${key}-${color}`} 
                                className={selected} 
                                style={{"background": color}}
                                onClick={() => onClick(color)}
                            >
                            </div>
                })}

                <h2>{title}</h2>
            </div>

}

export default Colors ;