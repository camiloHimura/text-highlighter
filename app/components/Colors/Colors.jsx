import React from 'react';
import PropTypes from "prop-types";
import {Container, Title} from "./styles.js"

function Colors(props){
  const {options, optionSelected, onClick, title = ''} = props;

  return  <div data-test='container' className="--flex --alignCenter">
              {options.map((color, key) => {
                  const selected = optionSelected === color ? 'selected': '';
                  
                  return <Container 
                            data-test='color'
                            className={selected}
                            key={`${key}-${color}`} 
                            style={{"background": color}}
                            onClick={() => onClick(color)}
                          >
                          </Container>
              })}

              <Title>{title}</Title>
          </div>

}

Colors.propTypes = {
    options: PropTypes.arrayOf(PropTypes.string).isRequired,
    optionSelected: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
    onClick: PropTypes.func.isRequired,
}

export default Colors ;
