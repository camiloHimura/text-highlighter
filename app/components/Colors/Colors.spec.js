import React from 'react';
import Colors from './Colors';
import { shallow } from 'enzyme'


describe('Colors component', () => {
  let Component;
  let props;

  beforeEach(() => {
    props = {
      onClick: jest.fn(),
      title: 'title test',
      options: ['blue', 'red', 'green'],
      optionSelected: 'optionSelected test',
    }
    Component = shallow(<Colors {...props}/>);
  })
  it('take snapshot', () => {
    expect(Component).toMatchSnapshot();
  });

  it('color rendered', () => {
    expect(Component.find('[data-test="color"]')).toHaveLength(props.options.length);
  });

  it('color applied', () => {
    Component.find('[data-test="color"]').forEach((node, index) => {
      expect(node.prop('style')).toEqual({"background": props.options[index]});
    })
  });

  it('color clicked', () => {
    const index = 0;
    Component.find('[data-test="color"]').at(index).prop('onClick')();
    expect(props.onClick).toHaveBeenCalledWith(props.options[index]);
  });


})