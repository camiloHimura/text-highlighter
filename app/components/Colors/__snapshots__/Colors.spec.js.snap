// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colors component take snapshot 1`] = `
<div
  className="--flex --alignCenter"
  data-test="container"
>
  <styled.div
    className=""
    data-test="color"
    key="0-blue"
    onClick={[Function]}
    style={
      Object {
        "background": "blue",
      }
    }
  />
  <styled.div
    className=""
    data-test="color"
    key="1-red"
    onClick={[Function]}
    style={
      Object {
        "background": "red",
      }
    }
  />
  <styled.div
    className=""
    data-test="color"
    key="2-green"
    onClick={[Function]}
    style={
      Object {
        "background": "green",
      }
    }
  />
  <styled.h2>
    title test
  </styled.h2>
</div>
`;
